File: src/main/java/org/spilth/validators/LanguageValidator.java
package org.spilth.validators;

import com.beust.jcommander.IParameterValidator;
import com.beust.jcommander.ParameterException;

import java.util.Arrays;
import java.util.List;

public class LanguageValidator implements IParameterValidator {
    public static final List<String> VALID_TYPES = Arrays.asList("java", "kotlin");

    @Override
    public void validate(String name, String value) throws ParameterException {
        if (!VALID_TYPES.contains(value)) {
            throw new ParameterException(String.format("Parameter %s needs to be either `java` or `kotlin`.", name));
        }
    }
}

File: src/main/kotlin/org/spilth/validators/LanguageValidator.kt
package org.spilth.validators

import com.beust.jcommander.IParameterValidator
import com.beust.jcommander.ParameterException
import java.util.*

class LanguageValidator : IParameterValidator {

    @Throws(ParameterException::class)
    override fun validate(name: String, value: String) {
        if (!VALID_TYPES.contains(value)) {
            throw ParameterException(String.format("Parameter %s needs to be either `java`, `java8`, or `kotlin`.", name))
        }
    }

    companion object {
        val VALID_TYPES = Arrays.asList("java", "java8", "kotlin")
    }
}
